source/include/MarlinUtil.h

```diff
@@ -61,6 +61,44 @@ namespace MarlinUtil {
   double getEnergyDepositedInFullCalorimeter(lcio::LCEvent* evt);
 
 
+  /** Return track weight contribution encoded as trackwgt = (int(wgt)%10000)/1000. by
+   * the MarlinReco/Analysis/RecoMCTruthLink/include/RecoMCTruthLinker.h for the
+   * ReconstructedParticle-MCParticle (or vise-versa) type of relations.
+   * 
+   */
+  inline float getTrackWeight(float encodedWeight){
+      return ( int(encodedWeight) % 10000 ) / 1000.;
```

Comment:
```suggestion
      return float( int(encodedWeight) % 10000 ) / 1000.f;
```
Just to not trigger a potentialconversion warnings, make all of the explicit.